element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,600 @@
+            electrode_group=nwbfile.electrode_groups[
+                f"probe{probe_id}_shank{shank_num}"
+            ],
+            cluster_quality_label=unit["cluster_quality_label"],
+            spike_times=unit["spike_times"],
+            spike_depths=unit["spike_depths"],
+            waveform_mean=waveform_mean,
+        )
+
+    return units_table
+
+
+def add_ephys_units_to_nwb(
+    session_key: dict, nwbfile: pynwb.NWBFile, primary_clustering_paramset_idx: int = 0
+):
+    """
+    Add spiking data to NWBFile.
+
+    ephys.CuratedClustering.Unit::unit -> units.id
+    ephys.CuratedClustering.Unit::spike_times -> units["spike_times"]
+    ephys.CuratedClustering.Unit::spike_depths -> units["spike_depths"]
+    ephys.CuratedClustering.Unit::cluster_quality_label -> units["cluster_quality_label"]
+
+    ephys.WaveformSet.PeakWaveform::peak_electrode_waveform -> units["waveform_mean"]
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    primary_clustering_paramset_idx: int, optional
+    """
+
+    if not ephys.ClusteringTask & session_key:
+        return
+
+    if nwbfile.electrodes is None:
+        add_electrodes_to_nwb(session_key, nwbfile)
+
+    # add additional columns to the units table
+    units_query = ephys.CuratedClustering.Unit() & session_key
+
+    for paramset_record in (
+        ephys.ClusteringParamSet & ephys.CuratedClustering & session_key
+    ).fetch("paramset_idx", "clustering_method", "paramset_desc", as_dict=True):
+        if paramset_record["paramset_idx"] == primary_clustering_paramset_idx:
+            units_table = create_units_table(
+                session_key,
+                nwbfile,
+                units_query,
+                paramset_record,
+                desc=paramset_record["paramset_desc"],
+            )
+            nwbfile.units = units_table
+        else:
+            name = f"units_{paramset_record['clustering_method']}"
+            units_table = create_units_table(
+                session_key,
+                nwbfile,
+                units_query,
+                paramset_record,
+                name=name,
+                desc=paramset_record["paramset_desc"],
+            )
+            ecephys_module = get_module(nwbfile, "ecephys")
+            ecephys_module.add(units_table)
+
+
+def get_electrodes_mapping(electrodes):
+    """
+    Create a mapping from the group (shank) and electrode id within that group to the row number of the electrodes
+    table. This is used in the construction of the DynamicTableRegion that indicates what rows of the electrodes
+    table correspond to the data in an ElectricalSeries.
+
+    Parameters
+    ----------
+    electrodes: hdmf.common.table.DynamicTable
+
+    Returns
+    -------
+    dict
+
+    """
+    return {
+        (electrodes["group"][idx].device.name, electrodes["id_in_probe"][idx],): idx
+        for idx in range(len(electrodes))
+    }
+
+
+def gains_helper(gains):
```

Comment:
Converting here because SpikeGLX and Open Ephys store data in units of microvolts and NWB stores data in volts?  Could you please add a quick doc string to this function?  Thanks.